petastorm/reader.py

```diff
@@ -284,6 +284,13 @@ def fetch(self, timeout=None):
         logger.warn(warning_message)
         return self._workers_pool.get_results(timeout=timeout)
 
+    def __len__(self):
```

Comment:
please add a unit test for this function

Reply:
Definitely. I added a new case to `test_end_to_end.py`.

Arguably, I should include a case to test raising `NotImplementedError`!